pennylane/grouping/utils.py

```diff
@@ -0,0 +1,589 @@
+        operation = None
+        if binary_vector[i] == 1 and binary_vector[n_qubits + i] == 0:
+            operation = PauliX(wires=label_map[i])
+
+        elif binary_vector[i] == 1 and binary_vector[n_qubits + i] == 1:
+            operation = PauliY(wires=label_map[i])
+
+        elif binary_vector[i] == 0 and binary_vector[n_qubits + i] == 1:
+            operation = PauliZ(wires=label_map[i])
+
+        if operation is not None:
+            if pauli_word is None:
+                pauli_word = operation
+            else:
+                pauli_word @= operation
+
+    if pauli_word is None:
+        return Identity(wires=list(label_map.values())[0])
+
+    return pauli_word
+
+
+def is_qwc(pauli_vec_1, pauli_vec_2):
+    """Checks if two Pauli words in the binary vector representation are qubit-wise commutative.
+
+    **Usage example:**
+
+    >>> is_qwc([1,0,0,1,1,0],[1,0,1,0,1,0])
+    False
+    >>> is_qwc([1,0,1,1,1,0],[1,0,0,1,1,0])
+    True
+
+    Args:
+        pauli_vec_1 (Union[list, tuple, array]): first binary vector argument in qubit-wise
+            commutator.
+        pauli_vec_2 (Union[list, tuple, array]): second binary vector argument in qubit-wise
+            commutator.
+
+    Returns:
+        bool: returns True if the input Pauli words are qubit-wise commutative, returns False
+            otherwise.
+
+    Raises:
+        ValueError: if the input vectors are of different dimension, if the vectors are not of even
+        dimension, or if the vector components are not strictly binary.
+
+    """
+
+    if isinstance(pauli_vec_1, (list, tuple)):
+        pauli_vec_1 = np.asarray(pauli_vec_1)
+    if isinstance(pauli_vec_2, (list, tuple)):
+        pauli_vec_2 = np.asarray(pauli_vec_2)
+
+    if len(pauli_vec_1) != len(pauli_vec_2):
+        raise ValueError(
+            "Vectors a and b must be the same dimension, instead got shapes {} and {}.".format(
+                np.shape(pauli_vec_1), np.shape(pauli_vec_2)
+            )
+        )
+
+    if len(pauli_vec_1) % 2 != 0:
+        raise ValueError(
+            "Symplectic vector space must have even dimension, instead got vectors of shape {}.".format(
+                np.shape(pauli_vec_1)
+            )
+        )
+
+    if not (
+        np.array_equal(pauli_vec_1, pauli_vec_1.astype(bool))
+        and np.array_equal(pauli_vec_2, pauli_vec_2.astype(bool))
+    ):
+        raise ValueError(
+            "Vectors a and b must have strictly binary components, instead got {} and {}".format(
+                pauli_vec_1, pauli_vec_2
+            )
+        )
+
+    n_qubits = int(len(pauli_vec_1) / 2)
+
+    for i in range(n_qubits):
+
+        a_x_i = pauli_vec_1[i]
+        a_z_i = pauli_vec_1[n_qubits + i]
+        b_x_i = pauli_vec_2[i]
+        b_z_i = pauli_vec_2[n_qubits + i]
+
+        if not (
+            ((a_x_i == a_z_i == 0) or (b_x_i == b_z_i == 0))
+            or ((a_x_i == b_x_i) and (a_z_i == b_z_i))
+        ):
```

Comment:
Might be worth renaming and factoring out parts of the conditions here to boost readability:
* `a_x_i`, `a_z_i`, e.g. to `first_vec_ith_qubit_paulix`, `first_vec_ith_qubit_pauliz`
* `first_vec_qubit_is_identity = (a_x_i == a_z_i == 0)`
* `if not (first_vec_qubit_is_identity or second_vec_qubit_is_identity or ...)`

or other descriptive names